vectordraw/vectordraw.py

```diff
@@ -0,0 +1,387 @@
+    """
+    An XBlock that allows course authors to define vector drawing exercises.
+    """
+
+    # Content
+    display_name = String(
+        display_name="Title (display name)",
+        help="Title to display",
+        default="Vector Drawing",
+        scope=Scope.content
+    )
+
+    description = String(
+        display_name="Description",
+        help="Exercise description displayed above vector drawing box",
+        default="",
+        multiline_editor="html",
+        resettable_editor=False,
+        scope=Scope.content
+    )
+
+    width = Integer(
+        display_name="Width",
+        help="The width of the board in pixels",
+        default=550,
+        scope=Scope.content
+    )
+
+    height = Integer(
+        display_name="Height",
+        help="The height of the board in pixels",
+        default=400,
+        scope=Scope.content
+    )
+
+    bounding_box_size = Integer(
+        display_name="Bounding box size",
+        help=(
+            "Defines the bounding box height of the graph area. "
+            "The bounding box width is calculated from the width/height ratio."
+        ),
+        default=10,
+        scope=Scope.content
+    )
+
+    axis = Boolean(
+        display_name="Show axis",
+        help="Show the graph axis",
+        default=False,
+        scope=Scope.content
+    )
+
+    show_navigation = Boolean(
+        display_name="Show navigation",
+        help="Show navigation arrows and zoom controls",
+        default=False,
+        scope=Scope.content
+    )
+
+    show_vector_properties = Boolean(
+        display_name="Show vector properties",
+        help="Show box detailing vector properties",
+        default=True,
+        scope=Scope.content
+    )
+
+    show_slope_for_lines = Boolean(
+        display_name="Show slope for lines",
+        help="If True, slope will be shown for line objects.",
+        default=False,
+        scope=Scope.content
+    )
+
+    add_vector_label = String(
+        display_name="Add vector label",
+        help="Label for button that allows to add vectors to the board",
+        default="Add Selected Force",
+        scope=Scope.content
+    )
+
+    vector_properties_label = String(
+        display_name="Vector properties label",
+        help="Label for box that displays vector properties",
+        default="Vector Properties",
+        scope=Scope.content
+    )
+
+    background_url = String(
+        display_name="Background URL",
+        help="URL for background image",
+        default="",
+        scope=Scope.content
+    )
+
+    background_width = Integer(
+        display_name="Background width",
+        help="Width of background image",
+        default=0,
+        scope=Scope.content
+    )
+
+    background_height = Integer(
+        display_name="Background height",
+        help="Height of background image",
+        default=0,
+        scope=Scope.content
+    )
+
+    vectors = String(
```

Comment:
Another thing to keep in mind:  If we decide to keep this field in its current form, we eventually want to turn this field into one of type `List` to make the JSON serialisation implicit and to model better what it is.  We can't do this now, since `List` fields aren't handled by the `StudioEditableXBlockMixin`.

Same comment hold for the `points` field.